server/src/main/java/keywhiz/service/resources/automation/v2/LogArgumentsImpl.java

```diff
@@ -0,0 +1,26 @@
+package keywhiz.service.resources.automation.v2;
+
+import com.google.common.collect.Streams;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogArgumentsImpl implements MethodInterceptor {
+  private static final Logger logger = LoggerFactory.getLogger(LogArguments.class);
+  public Object invoke(MethodInvocation invocation) throws Throwable {
+    Object[] parameters = Arrays.stream(invocation.getMethod().getParameters()).map(parameter -> parameter.isNamePresent()? parameter.getName() : parameter.getType()).toArray();
```

Comment:
Alternatively, I can just do `getParameters()`, which includes the type of the parameter as well. See below for an example log.

`keywhiz.service.resources.automation.v2.LogArguments: Function Name: createGroup, Function Arguments: [keywhiz.api.model.AutomationClient automationClient=AutomationClient{name=client}, keywhiz.api.automation.v2.CreateGroupRequestV2 request=CreateGroupRequestV2{name=group6, description=, metadata={}}]`